### PR TITLE
Add API documentation for Orca AutoEstimator

### PR DESCRIPTION
**Pull request by PeterQiu0516**
Add API documentation for Orca AutoEstimator.

### CODE REVIEW

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 1]
```diff
@@ -98,8 +105,8 @@ def from_keras(*,
 
     def fit(self,
             data,
-            epochs=1,
             validation_data=None,
+            epochs=1,
             metric=None,
```
Comment (TheaperDeng) [inline pyzoo/zoo/orca/automl/auto_estimator.py:110]: please change the param order back(as well as the order of the document)

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 2]
```diff
@@ -109,6 +116,35 @@ def fit(self,
             scheduler=None,
             scheduler_params=None,
             ):
+        """
+        Do necessary preparations for the engine
+        :param data: data for training
```
Comment (TheaperDeng) [inline pyzoo/zoo/orca/automl/auto_estimator.py:121]: now we have (just) support `data` and `validation_data` to be a pytorch data creator when model builder create a pytorch model, please describe this in the document.
Here is a reference
https://github.com/TheaperDeng/analytics-zoo/blob/6881650d1746fbe4f89275307adab1fece3a770c/pyzoo/zoo/automl/model/base_pytorch_model.py#L96-L103

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 3]
```diff
@@ -23,6 +23,13 @@ def __init__(self,
                  logs_dir="/tmp/auto_estimator_logs",
                  resources_per_trial=None,
                  name=None):
+        """
+        Constructor
+        :param model_builder: model builder for target model
+        :param logs_dir: local dir to save training results
+        :param resources_per_trial: resources for each trial
```
Comment (TheaperDeng) [inline pyzoo/zoo/orca/automl/auto_estimator.py:30]: a dict in the form of ....
reference:https://docs.ray.io/en/master/tune/api_docs/execution.html

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 4]
```diff
@@ -23,6 +23,13 @@ def __init__(self,
...
+        :param name: searcher name
```
Comment (TheaperDeng) [inline pyzoo/zoo/orca/automl/auto_estimator.py:31]: AutoEstimator name

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 5]
```diff
@@ -109,6 +116,35 @@ def fit(self,
             scheduler=None,
             scheduler_params=None,
             ):
+        """
+        Do necessary preparations for the engine
```
Comment (TheaperDeng) [inline pyzoo/zoo/orca/automl/auto_estimator.py:120]: automatically fit the model and search the best hyperparameters

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 6]
```diff
@@ -23,6 +23,13 @@ def __init__(self,
                  logs_dir="/tmp/auto_estimator_logs",
                  resources_per_trial=None,
                  name=None):
+        """
+        Constructor
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:27]: Create an AutoEstimator

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 7]
```diff
@@ -23,6 +23,13 @@ def __init__(self,
...
+        :param model_builder: model builder for target model
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:28]: A ModelBuilder instance, from which we will build the target model for hyper parameter optimization.

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 8]
```diff
@@ -23,6 +23,13 @@ def __init__(self,
...
+        :param logs_dir: local dir to save training results
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:29]: Defaults to ...

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 9]
```diff
@@ -109,6 +116,35 @@ def fit(self,
...
+        """
+        Do necessary preparations for the engine
+        :param data: data for training
+               Pandas Dataframe:
+                   a Pandas dataframe for training
+               Numpy ndarray:
+                   a tuple in form of (x, y)
+                        x: ndarray for training input
+                        y: ndarray for training output
+        :param validation_data: data for validation
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:128]: validation_data type should be the same as data.

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 10]
```diff
@@ -109,6 +116,35 @@ def fit(self,
...
+               Pandas Dataframe:
+                   a Pandas dataframe for validation
+               Numpy ndarray:
+                   a tuple in form of (x, y)
+                        x: ndarray for validation input
+                        y: ndarray for validation output
+        :param epochs: max epochs for training
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:135]: Max number of epochs to train in each trial. Defaults to 1. If you have also set metric_threshold, a trial will stop if either it has been optimized to the metric_threshold or it has been trained for {epochs} epochs.

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 11]
```diff
@@ -109,6 +116,35 @@ def fit(self,
...
+        :param metric: metric name
+        :param metric_threshold: a trial will be terminated when metric threshold is met
+        :param n_sampling: number of sampling
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:138]: Number of times to sample from the search_space. Defaults to 1. If hp.grid_search is in search_space, the grid will be repeated n_sampling of times. If this is -1, (virtually) infinite samples are generated until a stopping condition is met.

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 12]
```diff
@@ -109,6 +116,35 @@ def fit(self,
...
+        :param search_space: a dict for search space
+        :param search_alg: str, all supported searcher provided by ray tune
+               (i.e."variant_generator", "random", "ax", "dragonfly", "skopt",
+               "hyperopt", "bayesopt", "bohb", "nevergrad", "optuna", "zoopt" and
+               "sigopt")
+        :param search_alg_params: extra parameters for searcher algorithm
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:144]: except space, metric and mode.

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 13]
```diff
@@ -109,6 +116,35 @@ def fit(self,
...
+                        y: ndarray for training output
+        :param validation_data: data for validation
+               Pandas Dataframe:
+                   a Pandas dataframe for validation
+               Numpy ndarray:
+                   a tuple in form of (x, y)
+                        x: ndarray for validation input
+                        y: ndarray for validation output
+        :param epochs: max epochs for training
+        :param metric: metric name
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:136]: str

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 14]
```diff
@@ -109,6 +118,50 @@ def fit(self,
             scheduler=None,
             scheduler_params=None,
             ):
+        """
+        Automaically fit the model and search for the best hyperparameters.
+
+        :param data: data for training
+               Pandas Dataframe:
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:125]: :param data: train data. 
      If the AutoEstimator is created with from_torch, data can be a tuple of ndarrays or a function that takes a config dictionary as parameter and returns a PyTorch DataLoader.
      If the AutoEstimator is created with from_keras, data can be a tuple of ndarrays.
      If data is a tuple of ndarrays, it should be in the form of (x, y), where x is training input data and y is training target data.

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 15]
```diff
@@ -109,6 +118,50 @@ def fit(self,
...
+                   a Pandas dataframe for validation
+               Numpy ndarray:
+                   a tuple in form of (x, y)
+                        x: ndarray for validation input
+                        y: ndarray for validation output
+               Pytorch Data Creater:
+                   a Function which takes a config dict and returns a
+                   torch.utils.data.DataLoader. torch.Tensor should be generated from the
+                   dataloader.
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:148]: remove

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 16]
```diff
@@ -109,6 +118,50 @@ def fit(self,
             scheduler=None,
             scheduler_params=None,
             ):
+        """
+        Automaically fit the model and search for the best hyperparameters.
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:122]: automatically

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 17]
```diff
@@ -109,6 +118,37 @@ def fit(self,
...
+                ndarrays or a function that takes a config dictionary as parameter 
+                and returns a PyTorch DataLoader.
+            If the AutoEstimator is created with from_keras, data can be a tuple of 
+                ndarrays.
+            If data is a tuple of ndarrays, it should be in the form of (x, y), 
+                where x is training input data and y is training target data.
+        :param epochs: Max number of epochs to train in each trial. Defaults to 1.
+            If you have also set metric_threshold, a trial will stop if either it has been
+            optimized to the metric_threshold or it has been trained for {epochs} epochs.
+        :param validation_data: data for validation, should be the same type as data
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:135]: Validation data. Validation data type should be the same as train data.

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 18]
```diff
@@ -109,6 +118,37 @@ def fit(self,
...
+        :param metric: str, metric name, e.g. "mse"
```
Comment (yushan111) [inline pyzoo/zoo/orca/automl/auto_estimator.py:136]: String. The evaluation metric name to optimize.

File: pyzoo/zoo/orca/automl/auto_estimator.py [hunk 19]
```diff
@@ -127,6 +167,9 @@ def fit(self,
         self._fitted = True
 
     def get_best_model(self):
+        """
+        Return the best model found by the AutoEstimator
+        """
```
Comment (shane-huang) [inline pyzoo/zoo/orca/automl/auto_estimator.py:172]: add ":return" in the api docs.

**Review by yushan111 — APPROVED** [review]

**Comment by PeterQiu0516** [comment]
Jenkins passed: http://10.239.176.111:18888/job/ZOO-PR-Validation/5662/
